core/io/config_file.cpp

```diff
@@ -207,8 +207,12 @@ Error ConfigFile::_internal_save(Ref<FileAccess> file) {
 			file->store_string("\n");
 		}
 		if (!E.key.is_empty()) {
-			file->store_string("[" + E.key.replace("]", "\\]") + "]\n\n");
-		}
+			file->store_string("[" + E.key.replace("]", "\\]") + "]\n");
+
+			if (!E.value.is_empty()) {
+				file->store_string("\n"); 
```

Comment:
Remove trailing whitespace to fix CI:

```suggestion
				file->store_string("\n");
```